src/screens/Departures/QuayView.tsx

```diff
@@ -30,27 +35,52 @@ export default function QuayView({
   navigateToDetails,
   searchTime,
   setSearchTime,
+  showOnlyFavorites,
+  setShowOnlyFavorites,
   testID,
   stopPlace,
 }: QuayViewProps) {
+  const styles = useStyles();
+  const {favoriteDepartures} = useFavorites();
   const {state, refresh} = useQuayData(
     quay,
+    showOnlyFavorites,
     searchTime?.option !== 'now' ? searchTime.date : undefined,
   );
 
   const quayListData: SectionListData<Quay>[] = [{data: [quay]}];
 
+  const placeHasFavorites = hasFavorites(
+    favoriteDepartures,
+    stopPlace.id,
+    stopPlace.quays?.map((q) => q.id),
+  );
+
+  // If all favorites are removed while setShowOnlyFavorites is true, reset the
+  // value to false
+  useEffect(() => {
+    if (!placeHasFavorites) setShowOnlyFavorites(false);
```

Comment:
What do you mean by "that there were no changes"? Not sure we're on the same page here. This handles the case where a user removes their favorites while the favorite toggle is active. As shown here: 

https://user-images.githubusercontent.com/1774972/183638952-fb4ec3f9-c45e-4e97-bae7-fb68aa839eda.mp4

I think it's clear enough what happened?

Reply:
There can be cases where a bus only goes on the weekend and where weekend busses has different line id than on regular days. So if the first time you open the app you see a departure and mark it as a favorite, then going to Monday with "show only favorites" activated you'll get "no result" but in reality, there are busses that travel that journey pattern just under a different line id. It can look like there is no busses, but there are. Should we show a warning when we have favorites activated but get no result (but there are other results)? Something like

"Showing all busses as there were no favorite departures" etc

Reply:
Had a discussion with Christian on this now. Since it's less work to just update the message text, we landed on just showing a message along the lines of "We found no departures among your favourites. Disable ‘View favourite departures only’ to show all departures".